### PR TITLE
Fix SOFA regression checks when externalized (by using REGRESSION_REFERENCES_DIR  as fallback)

### PR DESCRIPTION
**Pull request by fredroy**
https://github.com/sofa-framework/Regression/pull/56 removed the usage of the env,variable REGRESSION_REFERENCES_DIR, and the consequence is that it was not possible to have the regression project in an other location than <SOFA>/applications/plugins,
This is problematic if you have your regression project in an other location.....

This PR uses the  env,variable REGRESSION_REFERENCES_DIR as a fallback if the given dir is not correct, and adds more messages (useful to check what is going on)

### CODE REVIEW

File: Regression_test/RegressionSceneList.inl [hunk 1]
```diff
@@ -106,6 +110,52 @@ void RegressionSceneList<T>::collectScenesFromList(const std::string& scenesDir,
...
+                    fullPathReferenceDir = listDir + "/" + refDirFromVar;
+                }
+                else // all cases are not compliant, quitting
+                {
+                    msg_error(msgHeader) << "The reference path does not exist, and the fallback REGRESSION_REFERENCES_DIR is set but erroneous.";
+                    return;
+                }
+            }
+        }
+        else
```
Comment (bakpaul) [inline Regression_test/RegressionSceneList.inl:146]: To be removed ?

**Review by bakpaul — CHANGES_REQUESTED** [review]

File: Regression_test/RegressionSceneList.inl (hunk 1 above)
Comment (bakpaul) [reply Regression_test/RegressionSceneList.inl:146]: Otherwise this is great ! :+1:

Comment (fredroy) [reply Regression_test/RegressionSceneList.inl:146]: Removing the else () ; // OK ?
I like to always have a else() with a if(),  especially with the comment (kind of explicity way to say "this is where everything is ok")

Comment (bakpaul) [reply Regression_test/RegressionSceneList.inl:146]: Ok, I understand. The compilo will remove it anyway, so if it increase readability that's ok !

**Pull request merged**